Nipadhye/firewall packet capture

nikhilpadhye1 opened the pull request:
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [X] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. NRP 170 - 9/23
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month. Next release, 9/23 NRP 170 release
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [X] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

MikhailTryakhov committed:
Adds base for updating Microsoft.Network from version stable/2022-01-01 to version 2022-05-01

MikhailTryakhov committed:
Updates readme

MikhailTryakhov committed:
Updates readme

MikhailTryakhov committed:
Updates API version in new specs and examples

MikhailTryakhov committed:
Updates API version in new specs and examples

dtuCloud committed:
add DisableTcpStateTracking on nic (#19734)

Co-authored-by: Dan Tu <dantu@microsoft.com>

dtuCloud committed:
add DisableTcpStateTracking on nic (#19734)

Co-authored-by: Dan Tu <dantu@microsoft.com>

shnaya434 committed:
Adding new VIP Swap APIs to support extension resource design (#19698)

* Adding new VIP Swap APIs to support extension resource design

* fixing validation checks

* minor fix

* Reverting changes

* Minor fix to add a new line

shnaya434 committed:
Adding new VIP Swap APIs to support extension resource design (#19698)

* Adding new VIP Swap APIs to support extension resource design

* fixing validation checks

* minor fix

* Reverting changes

* Minor fix to add a new line

dtuCloud committed:
fix description (#19987)

Co-authored-by: Dan Tu <dantu@microsoft.com>

dtuCloud committed:
fix description (#19987)

Co-authored-by: Dan Tu <dantu@microsoft.com>

nimaller committed:
Add BillingType Property to ExpressRoutePorts (#19932)

* change port json

* changes

nimaller committed:
Add BillingType Property to ExpressRoutePorts (#19932)

* change port json

* changes

jiaczh committed:
add Virtual network gateway policy group api (#19766)

* finish update on connection configuration references

* resolve reference error

* finish prettier

* resolve sematic error

* finish update change

jarango committed:
enablePrivateLinkFastPath property for VirtualNetworkGatewayConnection resource (#20050)

jarango committed:
enablePrivateLinkFastPath property for VirtualNetworkGatewayConnection resource (#20050)

sindhualuguvelli1 committed:
Support for Per Rule Actions in Application Gateway WAF (#20027)

* Adding changes for per rule actions

* Fixing spaces

* prettier fix

* Addressing PR comments

* Address review comments

sindhualuguvelli1 committed:
Support for Per Rule Actions in Application Gateway WAF (#20027)

* Adding changes for per rule actions

* Fixing spaces

* prettier fix

* Addressing PR comments

* Address review comments

jianqi-zhao committed:
Add ColoLocation to ExpressRoutePort result (#20033)

* Add ColoLocation to ExpressRoutePort result

* Fix automation code check error

* Merge from network-2022-05-01

* Revert "Merge from network-2022-05-01"

This reverts commit f0671a040b2bad684921f8f8ad4b53bb0f4b4a93.

Co-authored-by: Jianqi Zhao <jianzhao@microsoft.com>

jianqi-zhao committed:
Add ColoLocation to ExpressRoutePort result (#20033)

* Add ColoLocation to ExpressRoutePort result

* Fix automation code check error

* Merge from network-2022-05-01

* Revert "Merge from network-2022-05-01"

This reverts commit f0671a040b2bad684921f8f8ad4b53bb0f4b4a93.

Co-authored-by: Jianqi Zhao <jianzhao@microsoft.com>

mchakka committed:
DDoS Per IP SKU + Custom Policy Deprecation Swagger PR (#19985)

* initial changes

* example file changes

* new files for new apis

* adding a period

* changing reference

* prettier fixes

* ddos fix

* lint diff fixes

* more lint diff errors

* fixing operation id

* changing bool to string enum

* example changes

* adding 202s as these are long running ops

* examples

Co-authored-by: Manas Chakka <manaschakka@microsoft.com>

mchakka committed:
DDoS Per IP SKU + Custom Policy Deprecation Swagger PR (#19985)

* initial changes

* example file changes

* new files for new apis

* adding a period

* changing reference

* prettier fixes

* ddos fix

* lint diff fixes

* more lint diff errors

* fixing operation id

* changing bool to string enum

* example changes

* adding 202s as these are long running ops

* examples

Co-authored-by: Manas Chakka <manaschakka@microsoft.com>

bhavanabheem committed:
Adding UnhealthyThreshold property to LB health probes, and updated examples with new property (#20001)

bhavanabheem committed:
Adding UnhealthyThreshold property to LB health probes, and updated examples with new property (#20001)

Commit:
Property to enable/disable OCSP revocation check on client certificate (#20301)

* add new property to enable/disable OCSP revocation check on client certificate for MTLS

* revert breaking change of updating enum property

Commit:
Property to enable/disable OCSP revocation check on client certificate (#20301)

* add new property to enable/disable OCSP revocation check on client certificate for MTLS

* revert breaking change of updating enum property

gitlwh committed:
Add new parameter asn, geo and expressRouteAdvertise to CustomIPPrefix for new 2022-05-01 nrp sdk version (#20266)

* fix

* fix

* fix

* fix

* fix

* fix

Co-authored-by: Weiheng Li <weihl@microsoft.com>

gitlwh committed:
Add new parameter asn, geo and expressRouteAdvertise to CustomIPPrefix for new 2022-05-01 nrp sdk version (#20266)

* fix

* fix

* fix

* fix

* fix

* fix

Co-authored-by: Weiheng Li <weihl@microsoft.com>

jago2136 committed:
Network 2022-05-01 Azure Virtual Network Manager Contributions (#20440)

* Add third NIP enum for avnm security admin config, AllowRulesOnly

* add optional top parameters to various network manager post APIs

* revise all example input subscriptions to all zero guid

* change all top parameter max value to 100

* Revert "change all top parameter max value to 100"

This reverts commit 12943d2f2b91b22f1dae232cb291a8551fedfdca.

Co-authored-by: Jared Gorthy <jaredgorthy@microsoft.com>

jago2136 committed:
Network 2022-05-01 Azure Virtual Network Manager Contributions (#20440)

* Add third NIP enum for avnm security admin config, AllowRulesOnly

* add optional top parameters to various network manager post APIs

* revise all example input subscriptions to all zero guid

* change all top parameter max value to 100

* Revert "change all top parameter max value to 100"

This reverts commit 12943d2f2b91b22f1dae232cb291a8551fedfdca.

Co-authored-by: Jared Gorthy <jaredgorthy@microsoft.com>

alon-microsoft committed:
Alpelled waf manifest 05 01 2022 (#20047)

* init

* init

* fix comments

* fix err

* try ref to appgw def

* try ref to appgw def

* try ref to appgw def

* try ref to appgw def

* update example

* update example

* update example

* update state and action

* update state and action

* change ruleset name and add tiers

* cammel case fix

* cammel case fix

* s_ruleid

* s_RuleId

* suppress s_RuleId camel case issue

* suppress s_RuleId camel case issue

* address comments

* address comments

* address comments

* fix typo

* cont fix

* cont fix

* cont fix

* cont fix

* cont fix

* add pageable

* add next link

Co-authored-by: Alon Pelled <alpelled@microsoft.com>

alon-microsoft committed:
Alpelled waf manifest 05 01 2022 (#20047)

* init

* init

* fix comments

* fix err

* try ref to appgw def

* try ref to appgw def

* try ref to appgw def

* try ref to appgw def

* update example

* update example

* update example

* update state and action

* update state and action

* change ruleset name and add tiers

* cammel case fix

* cammel case fix

* s_ruleid

* s_RuleId

* suppress s_RuleId camel case issue

* suppress s_RuleId camel case issue

* address comments

* address comments

* address comments

* fix typo

* cont fix

* cont fix

* cont fix

* cont fix

* cont fix

* add pageable

* add next link

Co-authored-by: Alon Pelled <alpelled@microsoft.com>

nikhilpadhye1 committed:
Adding packet capture api for azure firewall

nikhilpadhye1 committed:
Adding packet capture api for azure firewall

nikhilpadhye1 committed:
Adding example file and correcting formatting

nikhilpadhye1 committed:
Adding example file and correcting formatting

nikhilpadhye1 committed:
Made changes as per API review and swagger validation failures

nikhilpadhye1 committed:
Made changes as per API review and swagger validation failures

nikhilpadhye1 committed:
Updating to align with swagger spec doc and prettier-fix ran

nikhilpadhye1 committed:
Updating to align with swagger spec doc and prettier-fix ran

lirenhe commented:
@nikhilpadhye1, please fix all the tool check errors.

lirenhe commented:
My understanding is network 2022-05-01 is already merged to main,  will you move your change to the next major relesae of NRP?

nikhilpadhye1 commented:
> 

Which release/api should i target? Our team is trying to get this change in this calendar year. Please let me know, the release schedule is a little confusing for me. 
I am targetting the NRP 171 release for my backend changes to support this new API. If you could provide some context here that would be greatly appreciated. 
Thanks a lot!

nikhilpadhye1 committed:
fixing semantic-validation error

nikhilpadhye1 committed:
fixing semantic-validation error

nikhilpadhye1 committed:
fixing lint-diff naming error

nikhilpadhye1 committed:
fixing lint-diff naming error

lirenhe commented:
> > 
> 
> Which release/api should i target? Our team is trying to get this change in this calendar year. Please let me know, the release schedule is a little confusing for me. I am targetting the NRP 171 release for my backend changes to support this new API. If you could provide some context here that would be greatly appreciated. Thanks a lot!

@MikhailTryakhov, could you provide the guide?

nikhilpadhye1 committed:
re-ran prettier check

nikhilpadhye1 committed:
re-ran prettier check

nikhilpadhye1 committed:
changing sasUrl naming to align with vpngateway api

nikhilpadhye1 committed:
changing sasUrl naming to align with vpngateway api

nikhilpadhye1 committed:
fixing lintDiff

nikhilpadhye1 committed:
fixing lintDiff

nikhilpadhye1 committed:
model validation fix

nikhilpadhye1 committed:
model validation fix

nikhilpadhye1 committed:
fixing lint-diff

nikhilpadhye1 committed:
fixing lint-diff

nikhilpadhye1 committed:
fixing model validation

nikhilpadhye1 committed:
fixing model validation

nikhilpadhye1 committed:
Fixing packet capture model to follow other api formats

nikhilpadhye1 committed:
Fixing packet capture model to follow other api formats

nikhilpadhye1 committed:
updating swagger for model validation errors

nikhilpadhye1 committed:
updating swagger for model validation errors

nikhilpadhye1 committed:
fixing lint-diff

nikhilpadhye1 committed:
fixing lint-diff

raosuhas commented on specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/azureFirewall.json:
```diff
@@ -1148,6 +1208,117 @@
+          "description": "Duration of packet capture in seconds."
+        },
+        "numberOfPacketsToCapture": {
+          "type": "string",
```
should these be int ?

nikhilpadhye1 replied:
yes, let me fix this. 
Will also need to move the PR to new api version. Will get this ARM review accepted and the nwill port over to new 2022-07-01 API

raosuhas commented:
signing off from ARM side with minor comments. Please fix before checking in.

raosuhas commented on specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/azureFirewall.json:
```diff
@@ -437,6 +437,66 @@
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously."
```
you also need to include 200 for final state of the location header

nikhilpadhye1 replied:
Is there anyway to only have 202 as response or is 200 mandatory for the reason above?
?

nikhilpadhye1 committed:
Merging 2022-07-01 api branch

nikhilpadhye1 committed:
Moving the PR to the new correct API version

nikhilpadhye1 committed:
Moving the PR to the new correct API version

nikhilpadhye1 commented:
@raosuhas, moved the API over to the new release 2022-07-01. Addressed the comments of having certain fields be intergers and adding 200 response code.

nikhilpadhye1 committed:
rebasing pr

nikhilpadhye1 committed:
merge errors
